Display number of runs selected for comparison

Before clicking on the `Compare` button on `/index.html`, display the number of runs that are going to be compared.

Inside of `Project Data` on `/compare.html` display the number of compared runs

@djesic @ArneTR I am not able to properly work with the VM setup which I have done. On my PC, I am getting the following error when I run install script. 

Do you guys have any solution for it. Setup on my PC would speed up the development process by many times. Thus asking.

```
Building binaries ...
Installing ./metric_providers/lm_sensors/metric-provider-binary ...
make: Entering directory '/home/saiteja/opensource/green-metrics-tool/metric_providers/lm_sensors'
gcc source.c chips.c -o3 -Wall -Llib -lsensors -lglib-2.0 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -o metric-provider-binary
source.c: In function ‘main’:
source.c:326:17: warning: implicit declaration of function ‘g_string_replace’; did you mean ‘g_tree_replace’? [-Wimplicit-function-declaration]
  326 |                 g_string_replace(chip_feature_str, " ", "-", 0);
      |                 ^~~~~~~~~~~~~~~~
      |                 g_tree_replace
/usr/bin/ld: /tmp/cc0Vs2qj.o: in function `main':
source.c:(.text+0xbcc): undefined reference to `g_string_replace'
collect2: error: ld returned 1 exit status
make: *** [Makefile:8: binary] Error 1
make: Leaving directory '/home/saiteja/opensource/green-metrics-tool/metric_providers/lm_sensors'
```

--- a/frontend/index.html
+++ b/frontend/index.html
@@ -56,7 +56,7 @@
                 <tr>
                     <th>Repositories (<i class="icon code github"></i> / <i class="icon code gitlab"></i> / <i class="icon code folder"></i> etc.)</th>
                     <th>Branch <i class="icon code branch"></i></th>
-                    <th><button id="compare-button" onclick="compareButton()" class="ui small button blue">Compare</button></th>
+                    <th><button id="compare-button" onclick="compareButton()" class="ui small button blue">Compare: 0 Run(s)</button></th>
                 </tr>
             </thead>
             <tbody></tbody>

--- a/frontend/index.html
+++ b/frontend/index.html
@@ -56,7 +56,7 @@
                 <tr>
                     <th>Repositories (<i class="icon code github"></i> / <i class="icon code gitlab"></i> / <i class="icon code folder"></i> etc.)</th>
                     <th>Branch <i class="icon code branch"></i></th>
-                    <th><button onclick="compareButton()" class="ui small button blue">Compare</button></th>
+                    <th><button id="compare-button" onclick="compareButton()" class="ui small button blue">Compare</button></th>
                 </tr>
             </thead>
             <tbody></tbody>

--- a/frontend/js/index.js
+++ b/frontend/js/index.js
@@ -9,10 +9,8 @@ const compareButton = () => {
 }
 const updateCompareCount = () => {
     const countButton = document.getElementById('compare-button');
-    const checkedCount = document.querySelectorAll('input#chbx-proj:checked').length;
-    countButton.textContent = checkedCount === 0 ? "Compare" 
-                              : checkedCount === 1 ?`Compare: ${checkedCount} Run` 
-                              : `Compare: ${checkedCount} Runs`;
+    const checkedCount = document.querySelectorAll('input[name=chbx-proj]:checked').length;
+    countButton.textContent = `Compare: ${checkedCount} Run(s)`;
 }
 
 function allow_group_select_checkboxes(checkbox_wrapper_id){
@@ -124,14 +122,14 @@ function allow_group_select_checkboxes(checkbox_wrapper_id){
             <td class="td-index" title="${filename}">${filename}</td>
             <td class="td-index">${machine}</td>
             <td class="td-index" style="width: 120px"><span title="${last_run}">${dateToYMD(new Date(last_run))}</span></td>
-            <td><input type="checkbox" value="${id}" name="chbx-proj" id="chbx-proj" />&nbsp;</td>`;
+            <td><input type="checkbox" value="${id}" name="chbx-proj"/>&nbsp;</td>`;
 
     });
 
     $('.ui.accordion').accordion();
     $('#projects-table table').tablesort();
 
-    document.querySelectorAll('#chbx-proj').forEach((e) =>{
+    document.querySelectorAll('input[name=chbx-proj]').forEach((e) =>{
         e.addEventListener('change', updateCompareCount);
     })
   

--- a/frontend/js/index.js
+++ b/frontend/js/index.js
@@ -7,6 +7,14 @@ const compareButton = () => {
     });
     window.location = link.substr(0,link.length-1);
 }
+const updateCompareCount = () => {
+    const countButton = document.getElementById('compare-button');
+    const checkedCount = document.querySelectorAll('input#chbx-proj:checked').length;
+    countButton.textContent = checkedCount === 0 ? "Compare" 
+                              : checkedCount === 1 ?`Compare: ${checkedCount} Run` 
+                              : `Compare: ${checkedCount} Runs`;
+}
+
 (async () => {
     const dateToYMD = (date) => {
         let day = date.getDate();
@@ -89,19 +97,23 @@ const compareButton = () => {
             <td class="td-index" title="${filename}">${filename}</td>
             <td class="td-index">${machine}</td>
             <td class="td-index" style="width: 120px"><span title="${last_run}">${dateToYMD(new Date(last_run))}</span></td>
-            <td><input type="checkbox" value="${id}" name="chbx-proj"/>&nbsp;</td>`;
+            <td><input type="checkbox" value="${id}" name="chbx-proj" id="chbx-proj" />&nbsp;</td>`;
 
     });
 
     $('.ui.accordion').accordion();
     $('#projects-table table').tablesort();
 
+    document.querySelectorAll('#chbx-proj').forEach((e) =>{
+        e.addEventListener('change', updateCompareCount);
+    })
 
     document.querySelectorAll('.toggle-checkbox').forEach((e) => {
         e.addEventListener('click', (e1) => {
             e1.currentTarget.closest('tr').querySelectorAll('td:first-child input').forEach((e2) => {
                 e2.checked = e1.currentTarget.checked
             })
+            updateCompareCount();
         })
     })
 

